src/engine.rs

```diff
@@ -147,6 +150,7 @@ impl From<NewCallArgs> for EngineState {
             owner_id: args.owner_id,
             bridge_prover_id: args.bridge_prover_id,
             upgrade_delay_blocks: args.upgrade_delay_blocks,
+            relayers_evm_addresses: LookupMap::new(KeyPrefix::RelayerEvmAddressMap),
```

Comment:
We don't need to store the key in storage. It's constant, so it's redundant and read/store same information to Engine state without need.

Reply:
TIL that `const generic` is available in rust, and use it for the first time to solve this, it was neat.